src/search/components/query-editor.controller.spec.js

```diff
@@ -490,4 +490,415 @@ describe('QueryEditorController', function() {
+      var result = controller.parseModalValuesFromQuery(query);
+      expect(result).toEqual(expectedResult);
+
+    });
+
+
+    it ('Should parse an advanced query', function(){
+
+      var $scope = $rootScope.$new();
+      var controller = $controller('QueryEditorController', { $scope: $scope });
+
+      var expectedResult = {
+        "type": "root",
+        "nodes": [
+          {
+            "type": "group",
+            "operator": "OR",
+            "nodes": [
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "streaming2020",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "virtueletoursmusea",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "uitpascoronaproof",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "workshopscoronaproof",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "liveonline",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "liveregistratie",
+                "name": "keywords",
+              },
+              {
+                "field": "labels",
+                "fieldType": "string",
+                "transformer": "=",
+                "term": "variacoronaproof",
+                "name": "keywords",
+              },
+              {
+                "type": "group",
+                "operator": "AND",
+                "nodes": [
+                  {
+                    "field": "dateRange",
+                    "fieldType": "date-range",
+                    "transformer": ">",
+                    "lowerBound": new Date("2020-02-29T23:00:00.000Z"),
+                    "upperBound": "*",
+                    "inclusive": true,
+                    "name": "date",
+                  }
+                ],
+              }
+            ],
+          }
+        ],
+        "operator": "AND"
```

Comment:
Seems like all the root operators are `AND`. Maybe add a test with an `OR` as root operator?

Reply:
Yes for some reason the root operator is always `AND`